pkgs/servers/home-assistant/appdaemon.nix

```diff
@@ -4,61 +4,62 @@ let
   python = python3.override {
     packageOverrides = self: super: {
       bcrypt = super.bcrypt.overridePythonAttrs (oldAttrs: rec {
-        version = "3.1.4";
+        version = "3.1.7";
         src = oldAttrs.src.override {
           inherit version;
-          sha256 = "13cyrnqwkhc70rs6dg65z4yrrr3dc42fhk11804fqmci9hvimvb7";
+          sha256 = "CwBpx1LsFBcsX3ggjxhj161nVab65v527CyA0TvkHkI=";
         };
       });
 
       yarl = super.yarl.overridePythonAttrs (oldAttrs: rec {
-        version = "1.1.0";
+        version = "1.4.2";
         src = oldAttrs.src.override {
           inherit version;
-          sha256 = "6af895b45bd49254cc309ac0fe6e1595636a024953d710e01114257736184698";
-        };
-      });
-
-      jinja2 = super.jinja2.overridePythonAttrs (oldAttrs: rec {
-        version = "2.10.1";
-        src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "065c4f02ebe7f7cf559e49ee5a95fb800a9e4528727aec6f24402a5374c65013";
-        };
-      });
-
-      aiohttp-jinja2 = super.aiohttp-jinja2.overridePythonAttrs (oldAttrs: rec {
-        version = "0.15.0";
-        src = oldAttrs.src.override {
-          inherit version;
-          sha256 = "0f390693f46173d8ffb95669acbb0e2a3ec54ecce676703510ad47f1a6d9dc83";
+          sha256 = "WM2cRp7O1VjNgao/SEspJOiJcEngaIno/yUQQ1t+90s=";
         };
       });
     };
   };
 
 in python.pkgs.buildPythonApplication rec {
   pname = "appdaemon";
-  version = "4.0.1";
+  version = "4.0.5";
 
   src = fetchFromGitHub {
     owner = "home-assistant";
     repo = "appdaemon";
     rev = version;
-    sha256 = "13qzjv11b0c7s1c66j70qmc222a78805n10lv2svj9yyk1v4xhjv";
+    sha256 = "7o6DrTufAC+qK3dDfpkuQMQWuduCZ6Say/knI4Y07QM=";
   };
 
   propagatedBuildInputs = with python.pkgs; [
     daemonize astral requests websocket_client aiohttp yarl jinja2
     aiohttp-jinja2 pyyaml voluptuous feedparser iso8601 bcrypt paho-mqtt setuptools
-    deepdiff dateutil bcrypt python-socketio pid
+    deepdiff dateutil bcrypt python-socketio pid pytz sockjs pygments
+    azure-mgmt-compute azure-mgmt-storage azure-mgmt-resource azure-keyvault-secrets azure-storage-blob
   ];
 
   # no tests implemented
   doCheck = false;
 
   postPatch = ''
-    substituteInPlace requirements.txt --replace "pyyaml==5.1" "pyyaml"
+    substituteInPlace requirements.txt \
+      --replace "pyyaml==5.3" "pyyaml" \
+      --replace "pid==2.2.5" "pid" \
+      --replace "Jinja2==2.11.1" "Jinja2" \
+      --replace "pytz==2019.3" "pytz" \
+      --replace "aiohttp==3.6.2" "aiohttp>=3.6" \
+      --replace "iso8601==0.1.12" "iso8601>=0.1" \
+      --replace "azure==4.0.0" "azure-mgmt-compute
+    azure-mgmt-storage
+    azure-mgmt-resource
+    azure-keyvault-secrets
+    azure-storage-blob" \
```

Comment:
we should let upstream know that the `azure` meta packages have been deprecated for more than a year, and your patch is the preferred method

Reply:
Opened an issue there, thank you